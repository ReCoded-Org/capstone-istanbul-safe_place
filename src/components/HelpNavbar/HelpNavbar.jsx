import React from "react";
import { Navbar, Nav, Form, Button, FormControl } from "react-bootstrap";
import phone from "../../images/icons/phone.svg";
import "./HelpNavbar.scss";

const HelpNavbar = () => (
  <Navbar className="help-navbar">
    <Nav className="mr-auto">
      <p>Immediate helpline:</p>
      <img src={phone} alt="phone" width="30px" />
      <p>+1-340-5558-235</p>
    </Nav>
    <Form inline>
      <FormControl type="text" placeholder="Search" className="mr-sm-2" />
      <Button variant="outline-primary">Search</Button>
    </Form>
  </Navbar>
);

export default HelpNavbar;
