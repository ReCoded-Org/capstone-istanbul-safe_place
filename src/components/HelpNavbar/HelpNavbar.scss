$PeachColor: #fe8f7c;
$LightCream: #fffefc;
$LightPeachColor: #ffbfb5;
$LightCreamOpacity: rgba(255, 254, 252, 0.6);

.help-navbar {
  background-color: $LightPeachColor;
  color: $LightCream;
  height: 3rem;
  padding: 0 14rem;
  p {
    margin: 0.1rem;
  }
  .form-control {
    height: 2rem;
  }
  button {
    height: 2rem;
    padding: 0 2rem;
    color: $PeachColor;
    background-color: $LightCreamOpacity;
    border-color: $LightCream;
  }
  button:focus {
    box-shadow: 0 0 0 0.2rem $PeachColor;
  }
  button:hover {
    color: #fff;
    background-color: $PeachColor;
    border-color: $PeachColor;
  }
  .form-control:focus {
    border-color: $PeachColor;
    box-shadow: 0 0 0 0.2rem $PeachColor;
  }
}
