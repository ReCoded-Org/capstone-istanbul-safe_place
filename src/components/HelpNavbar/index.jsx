import React from "react";
import {
  Navbar,
  Form,
  FormControl,
  Container,
  Col,
  Nav,
} from "react-bootstrap";
import phoneIcon from "../../images/icons/phone.svg";
import "./index.scss";

const HelpNavbar = () => (
  <Navbar className="helpNavbar">
    <Container>
      <Col>
        <p>Immediate helpline:</p>
        <img src={phoneIcon} alt="Phone icon" />
        <p>+1-340-5558-235</p>
      </Col>
      <Form inline>
        <FormControl
          type="text"
          placeholder="Find a shelter"
          className="mr-sm-2"
        />
        <Nav.Link href="/seekhelp">
          <button type="button">Search</button>
        </Nav.Link>
      </Form>
    </Container>
  </Navbar>
);

export default HelpNavbar;
