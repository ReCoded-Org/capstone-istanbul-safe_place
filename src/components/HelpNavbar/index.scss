@import "../../style/variables";
@import "../../style/buttons";
@import "../../style/bootstrapImports";

.helpNavbar {
  background-color: $warningColor;
  color: $lightColor;
  height: 2rem !important;

  .col {
    display: flex;
    flex-direction: row;
    margin-left: auto;
    justify-content: flex-end;
  }

  img {
    width: 1.2rem;
    margin: 0 0.6rem;
  }

  p {
    margin: 0.1rem 0;
  }

  a {
    color: $lightColor;
    text-decoration: underline;

    &:hover {
      font-weight: bold;
    }
  }
}

@include media-breakpoint-down(md) {
  .helpNavbar {
    height: 3rem !important;
  }
}
