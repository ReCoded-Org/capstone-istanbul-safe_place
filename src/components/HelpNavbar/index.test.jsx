import React from "react";
import renderer from "react-test-renderer";
import HelpNavbar from "./index";

it("renders correctly", () => {
  const tree = renderer.create(<HelpNavbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
