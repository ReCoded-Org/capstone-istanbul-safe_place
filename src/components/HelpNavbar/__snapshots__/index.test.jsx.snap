// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="helpNavbar navbar navbar-expand navbar-light"
>
  <div
    className="container"
  >
    <div
      className="col"
    >
      <p>
        helpNavbar.title
      </p>
      <img
        alt="Phone icon"
        src="phone.svg"
      />
      <a
        href="tel:+1-340-5558-235"
      >
        +1-340-5558-235
      </a>
    </div>
  </div>
</nav>
`;
